app/code/Magento/Catalog/Controller/Adminhtml/Product/MassDelete.php

```diff
@@ -54,16 +59,23 @@ public function __construct(
     }
 
     /**
+     * Mass Delete Action
+     *
      * @return \Magento\Backend\Model\View\Result\Redirect
      */
     public function execute()
     {
         $collection = $this->filter->getCollection($this->collectionFactory->create());
         $productDeleted = 0;
+        $productDeletedError = 0;
         /** @var \Magento\Catalog\Model\Product $product */
         foreach ($collection->getItems() as $product) {
-            $this->productRepository->delete($product);
-            $productDeleted++;
+            try {
+                $this->productRepository->delete($product);
+                $productDeleted++;
+            } catch (\Exception $exception) {
+                $productDeletedError++;
```

Comment:
As addition, I think that many developers can say 'thank you' if we can add logging or notification for the customer (as it is Localized exception) about reasons of the errors.

As I can see, for now (in 2.3-develop) it creates report in case of exception.